presentation/components/src/presentation-components/properties/InstanceKeyValueRenderer.tsx

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/**
+ * @packageDocumentation
+ * @module Properties
+ */
+
+import * as React from "react";
+import { Primitives, PrimitiveValue, PropertyRecord, PropertyValueFormat } from "@bentley/ui-abstract";
+import { IPropertyValueRenderer, PropertyValueRendererContext, TypeConverterManager, useAsyncValue } from "@bentley/ui-components";
+import { UnderlinedButton } from "@bentley/ui-core";
+import { translate } from "../common/Utils";
+import { useUnifiedSelectionContext } from "../unified-selection/UnifiedSelectionContext";
+
+/**
+ * Property value renderer for instance keys. If application provides a [[UnifiedSelectionContext]] and this value is
+ * clicked, it replaces current selection with the instance pointed by the key.
+ * @beta
+ */
+export class InstanceKeyValueRenderer implements IPropertyValueRenderer {
+  public canRender(record: PropertyRecord) {
+    return record.value.valueFormat === PropertyValueFormat.Primitive
+      && (record.value.value === undefined || isInstanceKey(record.value.value));
+  }
+
+  public render(record: PropertyRecord, context?: PropertyValueRendererContext) {
+    return <InstanceKeyValueRendererImpl record={record} context={context} />;
+  }
+}
+
+interface InstanceKeyValueRendererImplProps {
+  record: PropertyRecord;
+  context?: PropertyValueRendererContext;
+}
+
+const InstanceKeyValueRendererImpl: React.FC<InstanceKeyValueRendererImplProps> = (props) => {
+  const stringValue = useAsyncValue(convertRecordToString(props.record));
+  const valueElement = stringValue ?? props.context?.defaultValue;
+
+  const selectionContext = useUnifiedSelectionContext();
+  const instanceKey = (props.record.value as PrimitiveValue).value as Primitives.InstanceKey | undefined;
+
+  if (instanceKey === undefined || selectionContext === undefined) {
+    return <span style={props.context?.style} title={stringValue}>{valueElement}</span>;
+  }
+
+  const title = translate("instance-key-value-renderer.select-instance");
+  const handleClick = () => selectionContext.replaceSelection([instanceKey], 0);

```

Comment:
Should we use `selectionContext.selectionLevel` instead of `0` here?

Reply:
I don't think so. The navigation property might point to an instance that is not conceptually related to the same selection level.

Reply:
But by using `0` we give away the ability to control selection level from outside this component.  The property grid component will always want to change selection level at `0`, so this will work fine, but some other component might want to set this selection at a different level so the selection it uses as a "source" is not lost.

Reply:
Some property renderers in the future may want to interact with different selection levels, but having a single point of configuring the default level may not be that bad. Changed in 81416286eef63a3270ce9c3ad61dc38bbde64e24.